src/fabric.c

```diff
@@ -182,8 +178,15 @@ static void fi_ini(void)
 	free(liblist);
 done:
 #endif
+
+	/* these should be ordered by performance */
```

Comment:
Let's can this comment.  I really don't want to get into vendors arguing over which provider has better performance.  The order should be treated as arbitrary, with the exception of the sockets provider.  At best this change is working around bug(s) in the verbs provider and/or usnic verbs provider.

Also, is there a reason why you moved the static providers after the dynamic ones?  If so, can we add that to the commit message?